src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -339,6 +339,22 @@ def notify_query_listeners(self, query_record: QueryRecord) -> None:
         for listener in self._query_listener:
             listener._add_query(query_record)
 
+    def execute_and_get_sfqid(
+        self,
+        query: str,
+        is_ddl_on_temp_object: bool = False,
+        statement_params: Optional[Dict[str, str]] = None,
+    ) -> str:
+        if statement_params is None:
+            statement_params = {}
+        if is_ddl_on_temp_object:
+            statement_params["SNOWPARK_SKIP_TXN_COMMIT_IN_DDL"] = True
+        results_cursor = self._cursor.execute(query, _statement_params=statement_params)
+        self.notify_query_listeners(
+            QueryRecord(results_cursor.sfqid, results_cursor.query)
+        )
```

Comment:
Curious: is there a good way to reduce the duplication if `statement_params` and query listener's code? I would guess this code block is quite reusable.

Reply:
I think I can remove extra processing done on statement params are cursor is happy with accepting a None statement param. I'll remove ddl on temp object to keep things simple as it does not have any usage now. seeing other occurrences of statement params, does not seem like there is much duplication.

I'll create a helper for execute and notify query listener as I see them being used together frequently. Do you want me to refactor them in this PR or should I create a follow-up JIRA for that